### PR TITLE
fix(actions): update workflows, add auto-approval

### PR DESCRIPTION
**Pull request by raisedadead**
This PR updates the workflows to add one automatic approval by @camperbot for all renovate bot PRs, reducing the requirements to one additional reviewer with write access. It will also add a review request from me (for the moment, we can switch to dev-team if needed later). 

Secondly, I moved the i18n workflows to review requests from i18n team instead for transparency and as responsibilities shift within team. 

This when merged in tandem with #41590 should be a balance of notifications and turn around on a PR from staff.

### CODE REVIEW

File: .github/workflows/dependency-update.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - 'main'
+    paths:
+      - 'docs/index.html'
+      - '.github/workflows/*.yml'
+      - 'client/package.json'
+      - 'client/plugins/**/package.json'
+      - 'curriculum/package.json'
+      - 'package.json'
+      - 'tools/**/package.json'
+
```
Comment (raisedadead) [inline .github/workflows/dependency-update.yml:15]:
```suggestion
      # - 'api-server/**/package.json'
```

File: .github/workflows/dependency-update.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                pull_number: context.issue.number,
+                event: 'APPROVE',
+                body: "To be approved by `@freeCodeCamp/dev-team`"
+              });
+              github.pulls.requestReviewers({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                pull_number,: context.issue.number
```
Comment (nhcarrigan) [inline .github/workflows/dependency-update.yml:36]:
```suggestion
                pull_number: context.issue.number
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: .github/workflows/dependency-update.yml (hunk 2 above)
Comment (raisedadead) [reply .github/workflows/dependency-update.yml:36]: 🤦🏽

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me.

**Comment by raisedadead** [comment]
Hey @nhcarrigan! 

I was just going over this again and realized that the pr creation is implemented in an action by us. So, this bit here: 

https://github.com/freeCodeCamp/freeCodeCamp/blob/34dca74b503743610bb375633bda4ed6d46d4253/tools/crowdin/actions/pr-creator/index.js#L83

will not allow the changes I am making in this PR. It would need to accept a [`team_reviewers`](https://octokit.github.io/rest.js/v18#pulls-request-reviewers).

If it's the same with you I would like to drop the changes to the Crowdin bits in a later PR when I get around to updating the action.

**Comment by nhcarrigan** [comment]
That shouldn't be a problem - should be a quick variable rename from `reviewers` to `team_reviewers`.

**Comment by nhcarrigan** [comment]
I went ahead and shipped a potential fix.

File: .github/workflows/crowdin-i18n-client-ui-download.yml [hunk 3]
```diff
@@ -101,4 +101,4 @@ jobs:
           title: 'chore(i18n,client): processed translations'
           body: 'This PR was opened auto-magically by Crowdin.'
           labels: 'crowdin-sync, scope: UI'
-          reviewers: 'RandellDawson, nhcarrigan'
+          reviewers: '@freeCodeCamp/i18n'
```
Comment (nhcarrigan) [inline .github/workflows/crowdin-i18n-client-ui-download.yml:104]:
```suggestion
          team_reviewers: 'i18n'
```

File: .github/workflows/crowdin-i18n-curriculum-download.yml [hunk 4]
```diff
@@ -102,4 +102,4 @@ jobs:
           title: 'chore(i18n,curriculum): processed translations'
           body: 'This PR was opened auto-magically by Crowdin.'
           labels: 'crowdin-sync'
-          reviewers: 'RandellDawson, nhcarrigan'
+          reviewers: '@freeCodeCamp/i18n'
```
Comment (nhcarrigan) [inline .github/workflows/crowdin-i18n-curriculum-download.yml:105]:
```suggestion
          team_reviewers: 'i18n'
```

File: .github/workflows/crowdin-i18n-docs-download..yml [hunk 5]
```diff
@@ -135,4 +135,4 @@ jobs:
           title: 'chore(i18n,docs): processed translations'
           body: 'This PR was opened auto-magically by Crowdin.'
           labels: 'crowdin-sync'
-          reviewers: 'RandellDawson, nhcarrigan'
+          reviewers: '@freeCodeCamp/i18n'
```
Comment (nhcarrigan) [inline .github/workflows/crowdin-i18n-docs-download..yml:138]:
```suggestion
          team_reviewers: 'i18n'
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
These suggestions should match the changes in #41603

**Review by ahmaxed — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]
LGTM. Associated PRs are merged. Let's give this a shot.

**Pull request merged**